medusa/service/grpc/restore.py

```diff
@@ -65,9 +66,17 @@ def configure_console_logging(config):
 tables = {}
 use_sstableloader = False
 
-logging.info("Starting restore of backup {}".format(backup_name))
-
-medusa.restore_node.restore_node(config, tmp_dir, backup_name, in_place, keep_auth, seeds, verify, keyspaces, tables,
-                                 use_sstableloader)
-
-logging.info("Finished restore of backup {}".format(backup_name))
+cluster_backups = medusa.listing.get_backups(config, False)
+backup_found = False
+# Checking if the backup exists for the node we're restoring.
+# Skipping restore if it doesn't exist.
+for cluster_backup in cluster_backups:
+    if cluster_backup.name == backup_name:
+        backup_found = True
+        logging.info("Starting restore of backup {}".format(backup_name))
+        medusa.restore_node.restore_node(config, tmp_dir, backup_name, in_place, keep_auth,
+                                         seeds, verify, keyspaces, tables, use_sstableloader)
+        logging.info("Finished restore of backup {}".format(backup_name))
```

Comment:
I know that backup names are unique which means we should only get into this `if` block once. Would it good to add a `break` statement at the end after line 79?

Reply:
oooh yeah, most definitely 👍

Reply:
done